Tests/tests/test_TraceControl.cpp

```diff
@@ -0,0 +1,420 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+#include "gtest/gtest.h"
+#include <gmock/gmock.h>
+
+#include "TraceControl.h"
+
+#include "FactoriesImplementation.h"
+#include "IarmBusMock.h"
+#include "ServiceMock.h"
+#include "COMLinkMock.h"
+
+using namespace WPEFramework;
+
+namespace {
+const string callSign = _T("TraceControl");
+const string webPrefix = _T("/Service/TraceControl");
+const string volatilePath = _T("/tmp/");
+}
+
+class TraceControlTest : public ::testing::Test {
+protected:
+    Core::ProxyType<Plugin::TraceControl> plugin;
+    PluginHost::IWeb* interface;
+    Core::JSONRPC::Handler& handler;
+    Core::JSONRPC::Connection connection;
+    string response;
+
+    TraceControlTest()
+        : plugin(Core::ProxyType<Plugin::TraceControl>::Create())
+        , handler(*(plugin))
+        , connection(1, 0)
+    {
+        interface = static_cast<PluginHost::IWeb*>(plugin->QueryInterface(PluginHost::IWeb::ID));
+        Trace::TraceUnit::Instance().Open(volatilePath);
+    }
+
+    virtual ~TraceControlTest()
+    {
+        interface->Release();
+        plugin.Release();
+        Trace::TraceUnit::Instance().Close();
+    }
+};
+
+class TraceControlInitializedTest : public TraceControlTest {
+protected:
+    FactoriesImplementation factoriesImplementation;
+    IarmBusImplMock iarmBusImplMock;
+    ServiceMock service;
+    COMLinkMock comLinkMock;
+
+    TraceControlInitializedTest()
+        : TraceControlTest()
+    {
+    }
+    virtual ~TraceControlInitializedTest() override
+    {
+    }
+
+    void SetUp(bool bBackground)
+    {
+        ON_CALL(service, Background())
+            .WillByDefault(::testing::Return(bBackground));
+        ON_CALL(service, ConfigLine())
+            .WillByDefault(::testing::Return("{}"));
+        ON_CALL(service, WebPrefix())
+            .WillByDefault(::testing::Return(webPrefix));
+        ON_CALL(service, VolatilePath())
+            .WillByDefault(::testing::Return(volatilePath));
+        ON_CALL(service, Callsign())
+            .WillByDefault(::testing::Return(callSign));
+         ON_CALL(service, COMLink())
+             .WillByDefault(::testing::Return(&comLinkMock));
+
+        PluginHost::IFactories::Assign(&factoriesImplementation);
+        IarmBus::getInstance().impl = &iarmBusImplMock;
+        EXPECT_EQ(string(""), plugin->Initialize(&service));
+    }
+
+    virtual void TearDown()
+    {
+        plugin->Deinitialize(&service);
+        PluginHost::IFactories::Assign(nullptr);
+        IarmBus::getInstance().impl = nullptr;
+    }
+};
+
+TEST_F(TraceControlInitializedTest, registeredMethods)
+{
+    SetUp(true);
```

Comment:
please move the repeating code to the fixtures

Reply:
To reduce the repeated code, I added SetUp() with parameters, so that the tests can pass SetUp(true) & SetUp(false).
Let me know, which other repeated code can be moved?

Reply:
move
```
        ON_CALL(service, Background())
            .WillByDefault(::testing::Return(bBackground));
```
from fixture and add to the test
other things from SetUp(bool bBackground) move to TraceControlInitializedTest()
things from TearDown() move to ~TraceControlInitializedTest()
I hope this answers your question